examples/blake2f.rs

```diff
@@ -0,0 +1,1371 @@
+    [2, 12, 6, 10, 0, 11, 8, 3, 4, 13, 7, 5, 15, 14, 1, 9],
+    [12, 5, 1, 15, 14, 13, 4, 10, 0, 7, 6, 3, 9, 2, 8, 11],
+    [13, 11, 7, 14, 12, 1, 3, 9, 5, 0, 15, 4, 8, 6, 2, 10],
+    [6, 15, 14, 9, 11, 3, 0, 8, 12, 2, 13, 7, 1, 4, 10, 5],
+    [10, 2, 8, 4, 7, 6, 1, 5, 15, 11, 9, 14, 3, 12, 13, 0],
+];
+
+pub const XOR_VALUES: [u64; XOR_4BITS_NUMBER] = [
+    0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 1, 0, 3, 2, 5, 4, 7, 6, 9, 8, 11, 10, 13,
+    12, 15, 14, 2, 3, 0, 1, 6, 7, 4, 5, 10, 11, 8, 9, 14, 15, 12, 13, 3, 2, 1, 0, 7, 6, 5, 4, 11,
+    10, 9, 8, 15, 14, 13, 12, 4, 5, 6, 7, 0, 1, 2, 3, 12, 13, 14, 15, 8, 9, 10, 11, 5, 4, 7, 6, 1,
+    0, 3, 2, 13, 12, 15, 14, 9, 8, 11, 10, 6, 7, 4, 5, 2, 3, 0, 1, 14, 15, 12, 13, 10, 11, 8, 9, 7,
+    6, 5, 4, 3, 2, 1, 0, 15, 14, 13, 12, 11, 10, 9, 8, 8, 9, 10, 11, 12, 13, 14, 15, 0, 1, 2, 3, 4,
+    5, 6, 7, 9, 8, 11, 10, 13, 12, 15, 14, 1, 0, 3, 2, 5, 4, 7, 6, 10, 11, 8, 9, 14, 15, 12, 13, 2,
+    3, 0, 1, 6, 7, 4, 5, 11, 10, 9, 8, 15, 14, 13, 12, 3, 2, 1, 0, 7, 6, 5, 4, 12, 13, 14, 15, 8,
+    9, 10, 11, 4, 5, 6, 7, 0, 1, 2, 3, 13, 12, 15, 14, 9, 8, 11, 10, 5, 4, 7, 6, 1, 0, 3, 2, 14,
+    15, 12, 13, 10, 11, 8, 9, 6, 7, 4, 5, 2, 3, 0, 1, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3,
+    2, 1, 0,
+];
+
+fn split_value_4bits<F: PrimeField + Hash>(value: u128, n: usize) -> Vec<F> {
+    let mut value = value;
+    (0..n)
+        .map(|_| {
+            let v = value % VALUE_4BITS as u128;
+            value /= VALUE_4BITS as u128;
+
+            F::from(v as u64)
+        })
+        .collect()
+}
+
+fn split_xor_value<F: PrimeField + Hash>(value1: u64, value2: u64) -> Vec<F> {
+    let mut value1 = value1;
+    let mut value2 = value2;
+    let bit_values: Vec<u64> = (0..64)
+        .map(|_| {
+            let b1 = value1 % 2;
+            value1 /= 2;
+            let b2 = value2 % 2;
+            value2 /= 2;
+            b1 ^ b2
+        })
+        .collect();
+    (0..BITS_NUMBER)
+        .map(|i| {
+            F::from(
+                bit_values[i * 4]
+                    + bit_values[i * 4 + 1] * 2
+                    + bit_values[i * 4 + 2] * 4
+                    + bit_values[i * 4 + 3] * 8,
+            )
+        })
+        .collect()
+}
+
+fn blake2f_iv_table<F: PrimeField + Hash>(
+    ctx: &mut CircuitContext<F, ()>,
+    _: usize,
+) -> LookupTable {
+    let lookup_iv_row: Queriable<F> = ctx.fixed("iv row");
+    let lookup_iv_value: Queriable<F> = ctx.fixed("iv value");
+
+    let iv_values = IV_VALUES;
+    ctx.pragma_num_steps(IV_LENGTH);
+    ctx.fixed_gen(move |ctx| {
+        for (i, &value) in iv_values.iter().enumerate().take(IV_LENGTH) {
+            ctx.assign(i, lookup_iv_row, F::from(i as u64));
+            ctx.assign(i, lookup_iv_value, F::from(value));
+        }
+    });
+
+    ctx.new_table(table().add(lookup_iv_row).add(lookup_iv_value))
+}
+
+fn blake2f_4bits_table<F: PrimeField + Hash>(
```

Comment:
@10to4 I don't understand this table that both values are the same? Is it to see if the value is 4bits? In that case I would have a signal that is 1 always for all rows, and the other signal with `i`, Then you lookup to (1, x) always.

Reply:
Make sense. But I am not sure if all the keys are same, there will be potential problems in lookup table queries if we integrate more lookup schemes on Chiquito in the future. Because I'm not sure if all lookup table schemes allow the same key to exist.

Reply:
What you match are tuples, so it should be possible.